static/js/hash_util.js

```diff
@@ -270,3 +271,27 @@ export function active_stream() {
 export function is_create_new_stream_narrow() {
     return window.location.hash === "#streams/new";
 }
+
+export function is_web_public_compatible(hash) {
+    const web_public_allowed_hashes = [
+        "",
+        "narrow", // full #narrow hash handled in narrow.is_web_public_compatible
+        "recent_topics",
+        "keyboard-shortcuts",
+        "message-formatting",
+        "search-operators",
+        "all_messages",
+    ];
```

Comment:
I think `about_zulip` may belong here as well?  I wonder if it'd be better to do a denylist rather than an allowlist; is it just /#settings that's disabled?

Reply:
yeah but It should be better to have a list of hashes we allow rather than be uncertain about the hashes that a spectator can access.